logproof/src/linear_relation.rs

```diff
@@ -1404,4 +1409,256 @@ mod test {
     fn transcripts_match_k_4() {
         transcripts_match(4);
     }
+
+    fn convert_to_smallint(
```

Comment:
I don't quite understand what this function is doing. Subtracting modulus values is an odd thing to do. Maybe a code walkthrough tomorrow?

Reply:
Sure. Essentially for small negative numbers, the SEAL code that generates the RNS values essentially subtracts that small negative number from each moduli. So the idea here is just to convert back to the original small integers that make up `u` and `e` in order to  be able to encode them in any field (including a higher field) more easily.

Reply:
I've also added a temporary check that the small int conversion does not change the calculated answer. That test passes.

The main reason for it was to see if I was exporting values that made sense (ie were either ternary or were from the CBD with a std of 3.2)